app/src/main/kotlin/com/hippo/ehviewer/coil/MergeInterceptor.kt

```diff
@@ -14,7 +16,11 @@ object MergeInterceptor : Interceptor {
         val req = chain.request
         val key = req.memoryCacheKey?.takeIf { it.isNormalPreviewKey || Settings.preloadThumbAggressively }
         return if (key != null) {
-            mutex.withLock(key) { chain.proceed() }
+            val (result, suspended) = mutex.withLockNeedSuspend(key) { chain.proceed() }
+            when (result) {
+                is SuccessResult if (suspended) -> result.copy(dataSource = DataSource.DISK)
```

Comment:
If the first request was fulfilled by the memory cache, won't this make all requests animate except the first one?

Reply:
If we already have result in memory cache, then critical region is very small.
Actually it will not happen.